AIDevGallery/Samples/Open Source Models/Multimodal Models/DescribeImage.xaml.cs

```diff
@@ -153,68 +161,103 @@ private async IAsyncEnumerable<string> InferStreaming(string question, string im
+        }
+        else if (ButtonTextBlock.Text == "Cancel" && _cts != null)
+        {
+            CancelGeneration();
         }
     }
 
-    private async Task LoadAndDescribeImage(StorageFile file)
+    private void CancelGeneration()
     {
-        ButtonTextBlock.Text = "Cancel";
-        ToolTipService.SetToolTip(SelectImageButton, "Cancel generation");
-        AutomationProperties.SetName(SelectImageButton, "Cancel generation");
-        Output.Text = string.Empty;
-        Loader.IsActive = true;
-        Loader.Visibility = Visibility.Visible;
+        _cts!.Cancel();
+        DescribeImageButton.IsEnabled = false;
+        ButtonTextBlock.Text = "Canceling, this could take a while";
+    }
 
-        _cts = new CancellationTokenSource();
+    private async void LoadImage(StorageFile file)
+    {
+        using IRandomAccessStream fileStream = await file.OpenAsync(FileAccessMode.Read);
+        BitmapImage bitmapImage = new();
+        await bitmapImage.SetSourceAsync(fileStream);
+        DefaultImage.Source = bitmapImage;
+        NarratorHelper.AnnounceImageChanged(DefaultImage, "Image changed: new upload"); // <exclude-line>
+        imageFile = file;
+    }
 
-        using (IRandomAccessStream fileStream = await file.OpenAsync(FileAccessMode.Read))
+    private async Task DescribeTheImage(StorageFile file)
+    {
+        _cts = new CancellationTokenSource();
+        try
         {
-            BitmapImage bitmapImage = new();
-            await bitmapImage.SetSourceAsync(fileStream);
-            DefaultImage.Source = bitmapImage;
-            NarratorHelper.AnnounceImageChanged(DefaultImage, "Image changed: new upload"); // <exclude-line>
-        }
+            _cts.Token.ThrowIfCancellationRequested();
 
-        await Task.Run(async () =>
-        {
-            try
+            LoadImageButton.IsEnabled = false;
+            ButtonTextBlock.Text = "Cancel";
+            ToolTipService.SetToolTip(DescribeImageButton, "Cancel generation");
+            AutomationProperties.SetName(DescribeImageButton, "Cancel generation");
+            Output.Text = string.Empty;
+            Loader.IsActive = true;
+            Loader.Visibility = Visibility.Visible;
+
+            _cts.Token.ThrowIfCancellationRequested();
+
+            await Task.Run(async () =>
             {
-                await foreach (var part in InferStreaming("What is this image", file.Path, _cts.Token))
+                try
                 {
-                    DispatcherQueue.TryEnqueue(() => Output.Text += part);
+                    await foreach (var part in InferStreaming("What is this image", file.Path, _cts.Token))
+                    {
+                        DispatcherQueue.TryEnqueue(() => Output.Text += part);
+                    }
                 }
-            }
-            catch
-            {
-            }
-        });
-
-        ResetState();
+                catch
+                {
+                }
```

Comment:
Empty catch

Reply:
This has been here the whole time, idk if it works without it